### PR TITLE
feat(sbb-no-results): first implementation

### LINKED ISSUE #1927: story(sbb-no-results): initial implementation
### Feature Description

The 'no-result' component should show a message, for instance, if no results are found for a search.

Example already in usage (not lyne component yet): https://beta.sbb.ch/de?date=%222023-05-11%22&moment=%22DEPARTURE%22&stops=%5B%7B%22value%22%3A%228507364%22%2C%22type%22%3A%22ID%22%2C%22label%22%3A%22Jungfraujoch%22%7D%2C%7B%22value%22%3A%228507371%22%2C%22type%22%3A%22ID%22%2C%22label%22%3A%22M%C3%A4nnlichen%20%28Gondelbahn%29%22%7D%5D&time=%2209%3A07%22

[Figma Spec](https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?node-id=7577%3A292326&t=0zLXIu5HA8Iy8BxP-0)

**Design Spec**

- The component can be composed of several optional and non-optional elements in the following order:
  - An image which expands to 100% (?) --> (optional)
  - A title --> (non-optional)
  - A subtitle --> (non-optional)
  - An error code --> (optional)
  - A refresh button --> (optional)
- It expands to the `100%` of the container

**Technical Spec**

- The image should be provided via slot `name="image"` and should be an `sbb-image` (optional).
- The title should both be supported via slot `name="title"` (e.g. `<span slot="title">Title</span>`) and via property titleContent.
- The subtitle should be provided via slot `name="subtitle"` and needs to be a `<p>`.
- An additional label can be provided via slot `name="legend"` and needs to be a `<p>` (optional).
- `@Prop() titleContent?: string` to provide a title.
- `@Prop() public titleLevel?: InterfaceTitleAttributes['level'] = '3';` Level of title, will be rendered as heading tag (e.g. h3). Defaults to level 3.

```html
<sbb-no-results title-content="Title">
  <sbb-image slot="image" [...]></sbb-image>
  <p slot="subtitle">Subtitle</p>
  <p slot="legend">Error code: 0001</p>
  <sbb-button slot="action" [...]></sbb-button>
</sbb-no-results>
```

### Definition of Done

- [x] Component is implemented
- [x] Documentation is provided
- [x] Tests are implemented
- [x] E2E Tests are implemented
- [x] Storybook stories are implemented
- [ ] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [ ] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved

### PR DESCRIPTION
**Pull request by MarioCastigliano**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #1927 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- Implement sbb-no-results component

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [x] Chrome Mobile
- [x] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [x] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [x] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-no-results/sbb-no-results.tsx
Comment (dauriamarco) [inline src/components/sbb-no-results/sbb-no-results.tsx:49]: Sorry, I didn't notice before when we discussed it but the title logic should probably need to be adapted to the other similar implementations where we have the title slot inside the `sbb-title` itself (see `sbb-alert`and the `.sbb-alert__title` element for reference).

File: src/components/sbb-no-results/sbb-no-results.scss
Comment (dauriamarco) [inline src/components/sbb-no-results/sbb-no-results.scss:13]: Is the `display: flex` really needed? The `text-align: center` might be sufficient (?). And would it be possible to move the styles from the host to an inner container for consistency/stability? What do you think?

```scss
 .sbb-no-results__container {
  text-align: center;
}
```

File: src/components/sbb-no-results/sbb-no-results.scss
Comment (dauriamarco) [inline src/components/sbb-no-results/sbb-no-results.scss:19]: See comment about the title logic in `.tsx`

File: src/components/sbb-no-results/readme.md
Comment (dauriamarco) [inline src/components/sbb-no-results/readme.md:14]: See comment about the title logic in `.tsx`.

**Review by dauriamarco — CHANGES_REQUESTED** [review]
Hi and thanks a lot! Just few comments below :)

File: src/components/sbb-no-results/sbb-no-results.tsx
Comment (dauriamarco) [inline src/components/sbb-no-results/sbb-no-results.tsx:50]:
```suggestion
          <sbb-title level={this.titleLevel} visualLevel="5" class="sbb-no-results__title">
```

File: src/components/sbb-no-results/sbb-no-results.scss
Comment (dauriamarco) [inline src/components/sbb-no-results/sbb-no-results.scss:23]: To fix the title margin:
```suggestion
.sbb-no-results__title {
  // Overwrite sbb-title default margin
  margin: 0;
}
```

**Review by dauriamarco — CHANGES_REQUESTED** [review]

**Review by dauriamarco — APPROVED** [review]
LGTM 👍
We can wait for the UI review by @mcilurzo: https://lyne-components-preview-pr1943.app.sbb.ch/?path=/story/components-sbb-no-results--default

**Comment by mcilurzo** [comment]
ux-review: i found one small issue: https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj?node-id=26906:46319&mode=design#538204670

**Comment by MarioCastigliano** [comment]
@mcilurzo fixed, let me know if you find anything else https://lyne-components-preview-pr1943.app.sbb.ch/?path=/story/components-sbb-no-results--default

**Comment by mcilurzo** [comment]
@MarioCastigliano there are no more issues – grazie!

File: src/components/sbb-no-results/sbb-no-results.tsx
Comment (jeripeierSBB) [inline src/components/sbb-no-results/sbb-no-results.tsx:47]: Host element not needed

File: src/components/sbb-no-results/sbb-no-results.tsx
Comment (jeripeierSBB) [inline src/components/sbb-no-results/sbb-no-results.tsx:49]: checking with namedSlots should not be needed and can be removed

File: src/components/sbb-no-results/sbb-no-results.tsx
Comment (jeripeierSBB) [inline src/components/sbb-no-results/sbb-no-results.tsx:54]: checking with namedSlots should not be needed and can be removed

File: src/components/sbb-no-results/sbb-no-results.tsx
Comment (jeripeierSBB) [inline src/components/sbb-no-results/sbb-no-results.tsx:55]: checking with namedSlots should not be needed and can be removed

File: src/components/sbb-no-results/sbb-no-results.stories.tsx
Comment (jeripeierSBB) [inline src/components/sbb-no-results/sbb-no-results.stories.tsx:132]: nitpick, you could remove actions as its empty

File: src/components/sbb-no-results/sbb-no-results.scss
Comment (jeripeierSBB) [inline src/components/sbb-no-results/sbb-no-results.scss:40]: nice!

**Pull request merged**
